compat/visual-composer/visual-composer.php

```diff
@@ -0,0 +1,282 @@
+	}
+
+	function enqueue_active_widgets_scripts() {
+
+		global $wp_widget_factory;
+
+		foreach ( $wp_widget_factory->widgets as $class => $widget_obj ) {
+			if ( ! empty( $widget_obj ) && is_object( $widget_obj ) && is_subclass_of( $widget_obj, 'SiteOrigin_Widget' ) ) {
+				/* @var $widget_obj SiteOrigin_Widget */
+				ob_start();
+				$widget_obj->form( array() );
+				ob_clean();
+			}
+		}
+	}
+
+	function siteorigin_widget_form( $settings, $value ) {
+		$so_widget_names = array();
+
+		global $wp_widget_factory;
+
+		foreach ( $wp_widget_factory->widgets as $class => $widget_obj ) {
+			if ( ! empty( $widget_obj ) && is_object( $widget_obj ) && is_subclass_of( $widget_obj, 'SiteOrigin_Widget' ) ) {
+				$so_widget_names[ $class ] = $widget_obj->name;
+			}
+		}
+
+		/* @var $select SiteOrigin_Widget_Field_Select */
+		$select = new SiteOrigin_Widget_Field_Select(
+			'so_widget_class',
+			'so_widget_class',
+			'so_widget_class',
+			array(
+				'type'    => 'select',
+				'options' => $so_widget_names,
+			)
+		);
+
+		global $wp_widget_factory;
+
+		$parsed_value = json_decode( html_entity_decode( $value ), true );
+		if ( empty( $parsed_value ) ) {
+			//Get the first value as the default.
+			reset( $so_widget_names );
+			$widget_class = key( $so_widget_names );
+		} else {
+			$widget_class = $parsed_value['widget_class'];
+		}
+
+		$widget = ! empty( $wp_widget_factory->widgets[ $widget_class ] ) ? $wp_widget_factory->widgets[ $widget_class ] : false;
+
+		ob_start();
+		$select->render( $widget_class ); ?>
+		<input type="hidden" name="ajaxurl"
+		       data-ajax-url="<?php echo wp_nonce_url( admin_url( 'admin-ajax.php' ), 'sowb_vc_widget_render_form', '_sowbnonce' ) ?>">
+		<input type="hidden" name="so_widget_data" class="wpb_vc_param_value" value="<?php echo esc_attr( $value ); ?>">
+		<div class="siteorigin_widget_form_container">
+			<?php
+			if ( ! empty( $widget ) && is_object( $widget ) && is_subclass_of( $widget, 'SiteOrigin_Widget' ) ) {
+				/* @var $widget SiteOrigin_Widget */
+				$widget->form( $parsed_value['widget_data'] );
+			}
+			?>
+		</div>
+		<?php
+		return ob_get_clean();
+	}
+
+	function sowb_vc_widget_render_form() {
+		if ( empty( $_REQUEST['widget'] ) ) {
+			wp_die();
+		}
+		if ( empty( $_REQUEST['_sowbnonce'] ) || ! wp_verify_nonce( $_REQUEST['_sowbnonce'], 'sowb_vc_widget_render_form' ) ) {
+			wp_die();
+		}
+
+		$request      = array_map( 'stripslashes_deep', $_REQUEST );
+		$widget_class = $request['widget'];
+
+		global $wp_widget_factory;
+
+		$widget = ! empty( $wp_widget_factory->widgets[ $widget_class ] ) ? $wp_widget_factory->widgets[ $widget_class ] : false;
+
+		if ( ! empty( $widget ) && is_object( $widget ) && is_subclass_of( $widget, 'SiteOrigin_Widget' ) ) {
+			/* @var $widget SiteOrigin_Widget */
+			$widget->form( array() );
+		}
+
+		wp_die();
+	}
+
+	function update_widget_data( $content ) {
+
+		$content = preg_replace_callback( '/\[siteorigin_widget_vc [^\]]*\]/', array(
```

Comment:
This is totally a style preference, but I'd prefer to see this as one argument per line, where `array( $this, 'update_shortcode'  )` is a single argument. So:

```
$content = preg_replace_callback(
    '/\[siteorigin_widget_vc [^\]]*\]/', 
    array( $this, 'update_shortcode'),
    $content
);
```

To me, that's a lot more clear.